### PR TITLE
Add wallet edit command

### LINKED ISSUE #109: Add wallet edit command

### PR DESCRIPTION
**Pull request by gracegrape**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TRANSACTION_DISPLAYED_INDEX);
+        }
+
+        Transaction transactionToEdit = lastShownList.get(index.getZeroBased());
+        Transaction editedTransaction = createEditedTransaction(transactionToEdit, editTransactionDescriptor);
+
+        if (transactionToEdit instanceof Income) {
+
+            model.setIncome((Income) transactionToEdit, (Income) editedTransaction);
```
Comment (joel-lim) [inline src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java]: Maybe you could do it similarly to the delete command where you just have model.setTransaction, and let the underlying layer deal with the checking whether it is income or expense

**Review by joel-lim — COMMENTED** [review]
Looks good!

**Review by syinyichen — COMMENTED** [review]
Looks good! Thanks for the work!

File: src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the transaction identified "
+            + "by the index number used in the displayed transaction list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_NAME + "NAME "
+            + PREFIX_AMOUNT + "AMOUNT "
+            + "[" + PREFIX_DATE + "DATE] "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + WALLET_COMMAND_TYPE + " "
```
Comment (jumpcutfindo) [inline src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java:43]: Maybe the example can contain the optional information too for reference

File: src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+
+    private final Index index;
+    private final EditTransactionDescriptor editTransactionDescriptor;
+
+    /**
+     * @param index                     of the person in the filtered transaction list to edit
+     * @param editTransactionDescriptor details to edit the transaction with
+     */
+    public WalletEditCommand(Index index, EditTransactionDescriptor editTransactionDescriptor) {
+        requireNonNull(index);
```
Comment (jumpcutfindo) [inline src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java]: You can shorten this to `requireAllNonNull(obj1, obj2, ...)`

File: src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+        model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_TRANSACTIONS);
+        return new CommandResult(String.format(MESSAGE_EDIT_TRANSACTION_SUCCESS, editedTransaction));
+    }
+
+    /**
+     * Creates and returns a {@code Transaction} with the details of {@code transactionToEdit}
+     * edited with {@code editTransactionDescriptor}.
+     */
+    private static Transaction createEditedTransaction(Transaction transactionToEdit,
```
Comment (jumpcutfindo) [inline src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java:89]: I think you can afford to move this to the Transaction object, since modifying a Transaction might be a reusable method in the future

File: src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java (hunk 1 above)
Comment (gracegrape) [reply src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java]: Thank you for your suggestion! Agree with it.

File: src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java (hunk 2 above)
Comment (gracegrape) [reply src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java:43]: Okay! Thank you for your suggestion.

File: src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java (hunk 3 above)
Comment (gracegrape) [reply src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java]: Woah thanks! Didn't know about this.

File: src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java (hunk 4 above)
Comment (gracegrape) [reply src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java:89]: Hmmm, because the `PeopleEditCommnd` have their `createdEditedPerson` inside `PeopleEditCommnd` and not `People`.

Will it be better to keep both inside the respective EditCommand or should we move both out?

Comment (jumpcutfindo) [reply src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java:89]: In that case let's just keep it consistent. Thanks!
